ftdetect/hack.vim

```diff
@@ -0,0 +1,31 @@
+" Vim ftdetect file
+" Language:     Hack (PHP)
+" Maintainer:   Max Wang <mwang@fb.com>
+" URL:          https://github.com/hhvm/vim-hack
+" Last Change:  April 3, 2014
+"
+" Copyright: (c) 2014, Facebook Inc.  All rights reserved.
+"
+" This source code is licensed under the MIT license found in the
+" LICENSE file found at https://github.com/hhvm/vim-hack.
+"
+au BufRead,BufNewFile *.hack,*.hackpartial
+  \ setl filetype=hack |
+  \ setl syntax=hack
```

Comment:
I don't like that this mixes `filetype` and `syntax`. I believe the proper thing here is to make `hack` a subtype of `php`, as in `php.hack`?